processing/src/main/java/org/apache/druid/segment/LongDimensionIndexer.java

```diff
@@ -47,7 +48,7 @@ public class LongDimensionIndexer implements DimensionIndexer<Long, Long, Long>
   public EncodedKeyComponent<Long> processRowValsToUnsortedEncodedKeyComponent(@Nullable Object dimValues, boolean reportParseExceptions)
   {
     if (dimValues instanceof List) {
-      throw new UnsupportedOperationException("Numeric columns do not support multivalue rows.");
+      throw new ParseException(null, "Numeric columns do not support multivalue rows.");
     }
```

Comment:
is this check even necessary? `DimensionHandlerUtils.convertObjectToLong` will throw a parse exception for unknown types, same comment for other numeric indexers

Reply:
Yeah, I guess we can leave the check to `DimensionHandlerUtils.convertObjectToLong` and other equivalent methods. 

But in that case, we should have a proper error message there. Currently, `DimensionHandlerUtils` would just throw a `ParseException` with the message "Unknown type". 

Do you suggest we move the `instanceof List` check to `DimensionHandlerUtils.convertObjectToLong`?

Reply:
I think the exception should probably read "invalid type" or "incompatible type" instead of "unknown type". I guess my objection is that i'm not sure the multivalue rows part is actually that useful.

If i have my way there will _never_ be another multi-value anything in favor of using `ARRAY` types, so in that world these numeric value indexers will never support list or array or anything else other than stuff that can be converted to numbers, so I'm not sure its useful to specifically call out that "Numeric columns do not support multivalue rows".

Reply:
That makes sense. Thanks for the clarification!
I have updated the PR accordingly.

Reply:
I do think that its ok to have a special handling for list types so that errors are easier to make sense of. Right now, I will probably see below
```
could not convert [[2, 3, 4]] to long. Invalid type: java.util.ArrayList
```
a more readable message instead is 
```
could not ingest [2, 3, 4] since multiple values cannot be ingested for a long column in the same row. 
```

the latter looks more readable, to me at least.

Reply:
i don't feel that strongly about it so don't let me block this PR. I'm mostly just wondering what is special about the List type that it deserves its own error message.

If the input was from like say an array function transform it would be `Object[]` and not caught by the check.

If it were a list of map (like from list of json objects or whatever) so would be strings of that instead of numbers, and the error doesn't make quite as much sense either I would think. And so on, for other types.

Reply:
Users are most likely to run into this scenario where they are ingesting the data directly without transforms in between. IMO a more plain error message will users connect the dots faster than an error message that has java class names. Because the latter error message would appear as an internal issue on the face. 

If you are saying that we should be returning similar error messages for other types too, I agree. but in absence of that, we do the best we can. we could even add a utility method that can translate a java type classname to something more friendly (HashMap --> map).  and then folks can use it where they want.

Reply:
> we could even add a utility method that can translate a java type classname to something more friendly (HashMap --> map). and then folks can use it where they want.

I was thinking along similar lines but thought that it might be overkill. Hopefully having some clean error messages/error codes soon will help in these situations.

Reply:
>Users are most likely to run into this scenario where they are ingesting the data directly without transforms in between.

I guess that was just an example I was using to show that the space of things that can appear here is potentially very large. The indexers are entirely at the whims of whatever is underneath it is spitting out. We try to make things nice and tidy, but I can think of a couple very recent bug fixes which were about strange types ending up in places because of the behavior of the input format, #12828 and #12873.

>If you are saying that we should be returning similar error messages for other types too, I agree. but in absence of that, we do the best we can.

heh, I guess I was thinking the opposite, i don't think we can anticipate everything we could see here and so it seems strange to single out some subset with unique messages.

Seeing the value that wasn't a number seems like the most useful thing to have in the error message, since I can see that an array is obviously not a long, regardless of what the java type is (though the java type is important to be in the log message to help with bug reports).

>I was thinking along similar lines but thought that it might be overkill. Hopefully having some clean error messages/error codes soon will help in these situations.

It seems like overkill to me 😅

Reply:
ok. let's leave it to the PR author :)

Reply:
Merging this PR for now. We can improve the handling later when needed.